src/pages/tezos/transaction.js

```diff
@@ -136,12 +137,21 @@ function Transaction({ location }) {
   const [opHash, setOpHash] = useState(null);
   const [networkFeeEstimate, setNetworkFeeEstimate] = useState(0);
   const xtzPrice = location.state ? location.state.xtzPrice : null;
-  const bot = location.state ? location.state.bot : null;
+  const bot = location.state ? location.state.bot : false;
+  console.log('cryptobot object :', bot);
+  const [copyLink, setCopyLink] = useState(false);
   const { width, height } = useWindowSize();
   const [user, setUser] = useAtom(userAtom);
 
   const [claimButtonDisabled, setClaimButtonDisabledStatus] = useState(true);
 
+  useEffect(() => {
+    if (bot===false) {
+      console.log(window)
+      window.location.pathname="/tezos/marketplace";
+    }
+  }, []);
```

Comment:
Essentially I am redirecting user to the marketplace if the bot doesn't exist, it restricts user to directly g o to URL `/tezos/transaction` and see an empty screen.
Any better way to redirect?

Reply:
I think redirecting users here will confuse them. It's better to show an error & a back button to go to the previous page.